src/Bigint.php

```diff
@@ -7,17 +7,30 @@
 
 class Bigint
 {
+    /**
+     * @var array
+     */
     private $bytes = [0, 0, 0, 0, 0, 0, 0, 0];
 
+    /**
+     * Initialize the bytes array
+     *
+     * @param int $value
+     * @return void
+     */
     public function __construct(int $value = 0)
     {
-        if ($value instanceof self) {
-            $this->bytes = $value->bytes;
-        } else {
-            $this->fillBytes($value, 0, 8);
-        }
+        $this->fillBytes($value, 0, 8);
     }
 
+    /**
+     * Fill the bytes field with int
+     *
+     * @param int $value
```

Comment:
Why do we document param types & return types since they're already defined by PHP itself?

Reply:
(If we add more documentation to the argument like a detailed description, it makes total sense to me. But without that I can't see the benefit.)

Reply:
It helps various documentation generators or static analyzer to do their job properly. It also allows to add a comment to explain intent. Because it's nice to know the type, but if you don't know what or why you should have this param, it's less helpful. Doc comments are here also to be actual comments (not in this example though).

Reply:
I see how that helps in PHP 5 or earlier since there were no return type & no scalar type hints.

Reply:
But never mind, it also doesn't hurt.

Reply:
For instance it would be great to have a little comment explaining the purpose of $bytes in Bigint. And also a general comment on the class to know what it does. Because it's not obvious from the code :)